internal/command/workspace_command_test.go

```diff
@@ -16,11 +16,159 @@ import (
 	"github.com/hashicorp/terraform/internal/backend"
 	"github.com/hashicorp/terraform/internal/backend/local"
 	"github.com/hashicorp/terraform/internal/backend/remote-state/inmem"
+	"github.com/hashicorp/terraform/internal/providers"
 	"github.com/hashicorp/terraform/internal/states"
 	"github.com/hashicorp/terraform/internal/states/statefile"
 	"github.com/hashicorp/terraform/internal/states/statemgr"
 )
 
+func TestWorkspace_allCommands_pluggableStateStore(t *testing.T) {
```

Comment:
I wrote the E2E test for all the `workspace` commands and then worried that an integration test would be sufficient. After writing an integration test equivalent I realised that [the bug I found](https://github.com/hashicorp/terraform/pull/37855/files#r2504714904) was only discovered due to adding the E2E test, so the E2E test definitely has value being there.

At this point I'm unsure about whether to keep both tests or to only keep the E2E test.

Reply:
I'd be fine with E2E test coverage but the kind of E2E test we have now (`TestPrimary_stateStore_workspaceCmd`) makes it IMHO quite difficult to calculate code coverage because the testing interface is binary, rather than code, so unless you were planning on adding some lower-level E2E that does not have that downside I'd rather keep both.